tarsier/tag_utils.ts

```diff
@@ -458,67 +463,140 @@ function insertTags(
           (child as HTMLElement).tagName.toLowerCase(),
         )
       ) {
-        return getElementToInsertInto(child as HTMLElement, idSpan);
+        return getElementToInsertInto(child as HTMLElement);
       }
     }
 
     trimTextNodeStart(element);
-
-    // Base case
     return element;
   }
 
-  const idToTagMetadata: { [key: number]: TagMetadata } = {};
+  function getOpeningTag(el: HTMLElement): string {
+    const clonedEl = el.cloneNode(false) as HTMLElement;
+    const openingAndClosingTags = clonedEl.outerHTML;
+    const tagName = clonedEl.tagName.toLowerCase();
+    const closingTag = `</${tagName}>`;
+    if (openingAndClosingTags.endsWith(closingTag)) {
+      return openingAndClosingTags.slice(
+        0,
+        openingAndClosingTags.length - closingTag.length,
+      );
+    }
+    return openingAndClosingTags;
```

Comment:
I think this nesting can be cleaned up

<img width="1386" alt="image" src="https://github.com/user-attachments/assets/a891dc88-2887-4e5d-a48f-7dc1056273bf">